runtime/src/chpl-gpu.c

```diff
@@ -264,9 +311,33 @@ void* chpl_gpu_mem_alloc(size_t size, chpl_mem_descInt_t description,
   return ptr;
 }
 
+void* chpl_gpu_mem_array_alloc(size_t size, chpl_mem_descInt_t description,
```

Comment:
Great to see we that this is factoring out a lot of the common code from the AND and NVIDIA implementations.

I'm wondering if this is going to break things for `CHPL_GPU=cpu`, but I see that on your paratest list in the OP.

Reply:
Something certainly needs adjusting with that mode. I'll get to it and not worried about it at the moment.